Source/Renderer/modernizeShader.js

```diff
@@ -0,0 +1,233 @@
+        for (var a = 0; a < numLayoutVariables; ++a) {
+            var variableThatWeCareAbout = layoutVariables[a];
+            variableMap[variableThatWeCareAbout] = [null];
+        }
+
+        var stack = [];
+        for (var i = 0; i < splitSource.length; ++i) {
+            var line = splitSource[i];
+            var hasIF = /(#ifdef|#if)/g.test(line);
+            var hasELSE = /#else/g.test(line);
+            var hasENDIF = /#endif/g.test(line);
+
+            if (hasIF) {
+                stack.push(line);
+            } else if (hasELSE) {
+                var top = stack[stack.length - 1];
+                var op = top.replace('ifdef', 'ifndef');
+                if (/if/g.test(op)) {
+                    op = op.replace(/(#if\s+)(\S*)([^]*)/, '$1!($2)$3');
+                }
+                stack.pop();
+                stack.push(op);
+            } else if (hasENDIF) {
+                stack.pop();
+            } else if (!/layout/g.test(line)) {
+                for (var varIndex = 0; varIndex < numLayoutVariables; ++varIndex) {
+                    var varName = layoutVariables[varIndex];
+                    if (line.indexOf(varName) !== -1) {
+                        if (variableMap[varName].length === 1 && variableMap[varName][0] === null) {
+                            variableMap[varName] = stack.slice();
+                        } else {
+                            variableMap[varName] = variableMap[varName].filter(function(x) {
+                                return stack.indexOf(x) >= 0;
+                            });
+                        }
+                    }
+                }
+            }
+        }
+
+        for (var care in variableMap) {
+            if (variableMap.hasOwnProperty(care)) {
+                if (variableMap.length === 1 && variableMap[0] === null) {
+                    variableMap.splice(0, 1);
+                }
+            }
+        }
+
+        return variableMap;
+    }
+
+    function removeExtension(name, splitSource) {
+        var regex = '#extension\\s+GL_' + name + '\\s+:\\s+[a-zA-Z0-9]+\\s*$';
+        replaceInSourceRegex(new RegExp(regex, 'g'), '', splitSource);
+    }
+
+    /**
+     * A function to port GLSL shaders from GLSL ES 1.00 to GLSL ES 3.00
+     *
+     * This function is nowhere near comprehensive or complete. It just
+     * handles some common cases.
+     *
+     * Note that this function requires the presence of the
+     * "#define OUTPUT_DECLARATION" line that is appended
+     * by ShaderSource.
+     *
+     * @private
+     */
+    function modernizeShader(source, isFragmentShader) {
+        var outputDeclarationRegex = /#define OUTPUT_DECLARATION/;
+        var splitSource = source.split('\n');
+
+        if (/#version 300 es/g.test(source)) {
+            return source;
+        }
+
+        var outputDeclarationLine = -1;
+        var number;
+        for (number = 0; number < splitSource.length; ++number) {
```

Comment:
You probably got most of these in your latest commit - just double check but otherwise ignore these comments.